install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -1746,6 +1921,9 @@ nodePort:
 # default, always and never.
 # ref: https://docs.cilium.io/en/stable/security/policy/intro/#policy-enforcement-modes
 policyEnforcementMode: "default"
+# @schema
+# type: [null, string, array]
+# @schema
 # -- policyCIDRMatchMode is a list of entities that may be selected by CIDR selector.
 # The possible value is "nodes".
 policyCIDRMatchMode:
```

Comment:
This option seems to be defined as [`StringSlice`](https://github.com/cilium/cilium/blob/fc1f87bd2285778c7fb582fa23376ee8d52d61cd/daemon/cmd/daemon_main.go#L1069) in the daemon Go code. The helm template also uses `join` which usually operates on StringSlices. Sprig (templating functions library used by helm) casts the given string to a string slice with length 1. I guess thats the reason why both arrays and single strings seem to work in this case.

Validation wise this seems okay but might turn heads of curious readers.

Same goes for `ipam.operator.clusterPoolIPv4PodCIDRList` and possibly more

Reply:
It is a bit unfortunate, but I'm pretty sure that some users will have `clusterPoolIPv4PodCIDRList` set to a string, because it's predecessor value used to be a string too.

Ideally we'd want to deprecate the uses of strings here - but on the other hand, we want to be able to do it incrementally. It is a bit annoying that the Helm schema is "all or nothing" and that there is not a "emit a warning if a unsupported type or additional property is used" mode that would allow us to deprecate and fix those inconsistencies.

In any case, this seems fine as is for now.